cypress/e2e/homepage.spec.ts

```diff
@@ -0,0 +1,195 @@
+        .should("exist")
+        .contains(MOCK_NOTIFICATION_TEXT)
+    })
+
+    it("should only allow up to 4 highlights for the hero section", () => {
+      // Arrange
+      // NOTE: Toggle the dropdown
+      toggleSectionDropdown("Hero section")
+      cy.get(HERO_SECTION_HIGHLIGHTS_SELECTOR).check()
+      cy.contains("button", "Add highlight").as("addHighlightButton")
+
+      // Act
+      // NOTE: We assume that there are 3 highlights at the start
+      // as this is the initial state post-reset.
+      // NOTE: We require this because it clicks on the center by default
+      // which is where the text is.
+      // As we search for grandparent element to trigger the state change,
+      // clicking on the text will have the wrong grandparent element
+      // leading to no state change.
+      cy.get("@addHighlightButton").click("top")
+
+      // Assert
+      cy.get("@addHighlightButton").should("be.disabled")
+    })
+
+    it("should remove the button in preview when the hero button content is not given", () => {
+      // Arrange
+      const MOCK_HERO_BUTTON_TEXT = "This is a mock hero button text"
+      toggleSectionDropdown("Hero section")
+      cy.get(HERO_BUTTON_PREVIEW_SELECTOR)
+        .as("heroButtonPreview")
+        .should("exist")
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).clear().blur()
+      cy.get("@heroButtonPreview").should("not.exist")
+
+      // Act
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).type(MOCK_HERO_BUTTON_TEXT).blur()
+
+      // Assert
+      cy.get("@heroButtonPreview")
+        .should("exist")
+        .contains(MOCK_HERO_BUTTON_TEXT)
+    })
+
+    it("should preserve content when swapping between highlights and dropdown for hero section", () => {
+      // Arrange
+      const UPDATED_HERO_BUTTON_TEXT = "This is changed blah blah"
+      toggleSectionDropdown("Hero section")
+
+      // Act
+      cy.get(HERO_SECTION_HIGHLIGHTS_SELECTOR).check()
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).clear().type(UPDATED_HERO_BUTTON_TEXT)
+      cy.get(HERO_SECTION_DROPDOWN_SELECTOR).check()
+      cy.get(HERO_SECTION_HIGHLIGHTS_SELECTOR).check()
+
+      // Assert
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).should(
+        "have.value",
+        UPDATED_HERO_BUTTON_TEXT
+      )
+    })
+
+    it("should limit users to a single resource component", () => {
+      // Arrange
+      // NOTE: Precondition here is that there are NO existing resources
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.DEFAULT
+      )
+      cy.contains("div", "Add a new section")
+        .find("button")
+        .as("addSectionButton")
+        .should("be.disabled")
+
+      // Act
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.RESOURCES
+      )
+      cy.get("@addSectionButton").click()
+
+      // Assert
+      // NOTE: This is an existing bug on `production`
+      // where adding a resource section doesn't disable.
```

Comment:
are we planning to fix this? is there a ticket?

Reply:
Unsure, I think it's good to fix but it's difficult due to how messy this section of the code is...

Reply:
Hmm, then the test will keep failing right? Shall we comment this assert with a TODO on top?

ELse, it becomes like a legacy knowledge that oncalls need to know this failure is expected

Reply:
Sure, sounds food